packer/provisioner.go

```diff
@@ -162,6 +162,44 @@ func (p *PausedProvisioner) Provision(ctx context.Context, ui Ui, comm Communica
 	return p.Provisioner.Provision(ctx, ui, comm, generatedData)
 }
 
+// RetriedProvisioner is a Provisioner implementation that retries
+// the provisioner whenever there's an error.
+type RetriedProvisioner struct {
+	MaxRetries  int
+	Provisioner Provisioner
+}
+
+func (r *RetriedProvisioner) ConfigSpec() hcldec.ObjectSpec { return r.ConfigSpec() }
+func (r *RetriedProvisioner) FlatConfig() interface{}       { return r.FlatConfig() }
+func (r *RetriedProvisioner) Prepare(raws ...interface{}) error {
+	return r.Provisioner.Prepare(raws...)
+}
+
+func (r *RetriedProvisioner) Provision(ctx context.Context, ui Ui, comm Communicator, generatedData map[string]interface{}) error {
+	err := r.Provisioner.Provision(ctx, ui, comm, generatedData)
+	if err == nil {
+		return nil
+	}
+
+	leftTries := r.MaxRetries
+	for ; leftTries > 0; leftTries-- {
+		if ctx.Err() != nil { // context was cancelled
```

Comment:
Do you think it would be nice to check the context before the first run too ? Chances are low that this happens but still there. 🤔

Reply:
I don't think it won't change the behavior 🤔 this will be checked before retrying anyways, it will basically be two sequential checks.

Reply:
This changes the behaviour if the context is cancelled before the first run

Reply:
So you mean to check for cancellation as a first thing to do? Because then I think this check should be inside every provisioner because we need to check before running it with or without the retry thing 🤔  
Am I thinking right?

Reply:
The other wrappers don't check for cancellation. Should I add it to them as well?

Reply:
Done ✅